cmd/experimental/kjobctl/apis/v1alpha1/volumebundle_types.go

```diff
@@ -17,11 +17,37 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type MountPoint struct {
+	// Inlines Container’s volume mounts.
+	corev1.VolumeMount `json:",inline"`
+
+	// Environment variables pointing to locations on the mount.
+	// These values will be copied to pods/containers environment variables.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+}
+
 // VolumeBundleSpec defines the desired state of VolumeBundle
-type VolumeBundleSpec struct{}
+// +kubebuilder:validation:XValidation:rule="self.volumes.size() == self.mountPoints.size()", message="volumes and mountPoints must have the same amount of items"
```

Comment:
Not necessary, you could mount a volume (or pats of it) in multiple points. Ideally we should make sure that  `MountPoints[x].Name` is found in `Volumes[...].Name`